x-pack/test/api_integration/deployment_agnostic/default_configs/stateful.config.base.ts

```diff
@@ -98,6 +105,17 @@ export function createStatefulTestConfig<T extends DeploymentAgnosticCommonServi
         ...xPackAPITestsConfig.get('kbnTestServer'),
         serverArgs: [
           ...xPackAPITestsConfig.get('kbnTestServer.serverArgs'),
+          // if the config is run locally, explicitly enable mock-idp-plugin for UI role selector
+          ...(isRunOnCI ? [] : ['--mockIdpPlugin.enabled=true']),
+          // This ensures that we register the Security SAML API endpoints.
+          // In the real world the SAML config is injected by control plane.
+          `--plugin-path=${samlIdPPlugin}`,
+          '--xpack.cloud.id=ftr_fake_cloud_id',
+          // Ensure that SAML is used as the default authentication method whenever a user navigates to Kibana. In other
+          // words, Kibana should attempt to authenticate the user using the provider with the lowest order if the Login
+          // Selector is disabled (replicating Serverless configuration). By declaring `cloud-basic` with a higher
+          // order, we indicate that basic authentication can still be used, but only if explicitly requested when the
+          // user navigates to `/login` page directly and enters username and password in the login form.
```

Comment:
This part is copied from serverless config and probably needed for security APIs testing in deployment-agnostic tests. If I'm wrong, we can remove it

Reply:
@jeramysoucy you added similar arguments to serverless FTR config in #162087, I copied it for the consistency but please confirm if we need that.

Reply:
Are you referring specifically to `xpack.cloud.id`? For stateful I don't think we need `xpack.cloud.id`. In serverless, certain interfaces/capabilities will not be present unless there is a cloud ID.

Reply:
I think the mock IDP plugin is enough, but perhaps I am missing something. cc @azasypkin